apm/src/app/user/user.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { IUser } from './user';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UserService {
+
+  currentUser: IUser | undefined;
+    
+  login(userName: string, password: string) {
+    this.currentUser = {
+      id : 1,
+      firstName: userName,
+      lastName: ''
+    }
+  }
+
+  isAuthenticated() {
+    return !!this.currentUser;
```

Comment:
I have used this myself, but over the years the suggestion that I have received is to use `Boolean(this.currentUser)`